core/src/parser/mod.rs

```diff
@@ -154,12 +154,22 @@ impl<'ast> Parser<'ast> {
 
     #[inline]
     fn parameter_list(&mut self) -> ParameterList<'ast> {
+        let mut default_params = false;
         let first = match self.next() {
             Identifier(label) => {
-                self.in_loc(Parameter::Identifier {
+                let parameter = Parameter::Identifier {
                     label,
-                    value: None
-                })
+                    value: match self.peek() {
```

Comment:
I think this can be reworked to use `self.next()` instead of `self.peek()`, since there are only 3 tokens that can follow an identifier in parameter list: `Operator(Assign)`, `Comma` and `ParenClose`. Peeking is expensive* as it means the same token is going to be read twice (or more), and forces a memmove on the next token.

* Expensive given we are in 2 nanoseconds per byte parsed range :).